src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/SystemDiagnosticsMetricsTrigger/SystemDiagnosticsMetricsTrigger.cs

```diff
@@ -67,11 +70,19 @@ public static MetricSourceConfiguration CreateConfiguration(SystemDiagnosticsMet
         {
             Validate(settings);
 
+            bool useSharedSession = false;
```

Comment:
For correctness, I would further pull this out and add a `UseSharedSession` property to `SystemDiagnosticsMetricsTriggerSettings`; let the caller ultimately decided whether shared sessions are supported.

Reply:
Resolving within the Dotnet Monitor PR